lib/timers.js

```diff
@@ -47,6 +47,8 @@ const { kInit, kDestroy, kAsyncIdCounter } = async_wrap.constants;
 const async_id_symbol = Symbol('asyncId');
 const trigger_async_id_symbol = Symbol('triggerAsyncId');
 
+const functionApply = Function.prototype.apply;
```

Comment:
I'm not sure we can safely do this. This was explicitly avoided in https://github.com/nodejs/node/commit/98609fc1c4635f02f8f6ef9dd079207a1204b9a1.

Reply:
Yeah, so the question here is whether we want to pick up a monkey patched version of apply or not? I mean, my instinct says 'no' (and part of the reason for this adjustment) because realistically the internals could change again in the future to where `apply` is not even used anymore.

Reply:
/cc @nodejs/tsc

Reply:
I think the answer here might actually be different and that is, use `Reflect.apply` — assuming performance holds up.

Reply:
Switched to `Reflect.apply` which seems correct here. Quick test revealed no differences in performance. Open to other thoughts on this though.